x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/sequence/TumblingWindow.java

```diff
@@ -122,7 +122,10 @@ public TumblingWindow(
     public void execute(ActionListener<Payload> listener) {
         log.trace("Starting sequence window w/ fetch size [{}]", windowSize);
         startTime = System.currentTimeMillis();
-        tumbleWindow(0, listener);
+        tumbleWindow(0, wrap(listener::onResponse, e -> {
```

Comment:
I prefer the close method to be called in one central place - right now it's already called in two places and this fixes adds a third one for the exception branch.
There should be only one call, even removing the method itself to avoid having it reused.
Something along the lines of:
```
tumbleWindow(0, runAfter(listener, close(listener));
```

This should guarantee that after the listener executes, the clean-up is done both on onResponse and onFailure. It's worth checking whether onFailure can be called multiple times and if not, change close() method to not forward the exception.
Also I would remove close(listener) method and simply inline the code in the lambda above to avoid calling it from different places since that shouldn't be the case.